Utilities/O2device/include/O2device/O2device.h

```diff
@@ -0,0 +1,67 @@
+/********************************************************************************
+ *    Copyright (C) 2014 Frankfurt Institute for Advanced Studies               *
+ *                                                                              *
+ *              This software is distributed under the terms of the             *
+ *         GNU Lesser General Public Licence version 3 (LGPL) version 3,        *
+ *                  copied verbatim in the file "LICENSE"                       *
+ ********************************************************************************/
+/**
+ * O2device.h
+ *
+ * @since 2014-12-10
+ * @author M. Krzewicki <mkrzewic@cern.ch>
+ */
+
+#ifndef O2DEVICE_H_
+#define O2DEVICE_H_
+
+#include "FairMQDevice.h"
+#include "Headers/DataHeader.h"
+#include <stdexcept>
+
+namespace AliceO2 {
+namespace Base {
+
+class O2device : public FairMQDevice
+{
+public:
+  using FairMQDevice::FairMQDevice;
+  virtual ~O2device() {}
+
+  /// Here is how to add an annotated data part (with header);
+  /// second argument should be used
+  bool AddMessage(FairMQParts& parts, AliceO2::Header::Block&& incomingBlock, FairMQMessagePtr dataMessage) {
```

Comment:
I would not implement this message handling functionality directly in the class, but as a separate tool. Then the developer is free to use it somewhere else as well.